### PR TITLE
:book: Verifying the Authenticity and Integrity of Scorecard Results

### PR DESCRIPTION
**Pull request by naveensrinivasan**
- Included section about `Verifying the Authenticity and Integrity of Scorecard Results`
- Included a mermaid diagram.

Signed-off-by: naveensrinivasan <172697+naveensrinivasan@users.noreply.github.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -254,3 +256,23 @@ jobs:
...
+Scorecard action ->> Scorecard action: Generate result.json
+Note over cosign, Scorecard action: Scorecard  embeds the cosign library
+cosign ->> rekor.sigstore.dev: Sign and store hash of result.json 
+Scorecard action->>api.securityscorecards.dev: Post result.json
+Note over api.securityscorecards.dev: Validates the result
+api.securityscorecards.dev->>Scorecard action: Return status  
+```
+1. The scorecard action is a tool that runs a scorecard analysis on a repository.
+1. After running the scorecard analysis on the repository, the action generates a file called result.json that contains the analysis results.
+1. The cosign library, which is embedded in the scorecard action, is then used to sign and store hash of result.json file in the `rekor.sigstore.dev` service.
```
Comment (olivekl) [inline README.md:275]:
```suggestion
1. The cosign library, which is embedded in the Scorecard action, signs and stores a hash of result.json file in the `rekor.sigstore.dev` service.
```

File: README.md [hunk 2]
```diff
@@ -254,3 +256,23 @@ jobs:
...
+1. Finally, the result.json file is posted to the `api.securityscorecards.dev` service.
```
Comment (olivekl) [inline README.md:276]:
```suggestion
1. The result.json file is posted to the `api.securityscorecards.dev` service.
```

Comment (olivekl) [inline README.md:276]:
```suggestion
1. The result.json file is posted to the `api.securityscorecards.dev` service.
```

File: README.md (hunk 1 above)
Comment (olivekl) [reply README.md:275]: Is Sigstore / Cosign / Rekor discussed elsewhere? If not, this step lacks context for the reader.

File: README.md [hunk 3]
```diff
@@ -254,3 +256,23 @@ jobs:
...
+1. The `api.securityscorecards.dev` service validates the authenticity of the results with `rekor.sigstore.dev` before storing the results.
+1. This process allows the repository owner to verify the authenticity and integrity of the scorecard results and make them available for others to access and use.
```
Comment (olivekl) [inline README.md:278]: This is not a step in the process and doesn't below in the list. I suggest moving it to just before the diagram and list (after the sentence "This digram illustrates how Scorecard validates..."

**Review by olivekl — CHANGES_REQUESTED** [review]
Just a few suggestions!

File: README.md (hunk 3 above)
Comment (naveensrinivasan) [reply README.md]: Took care of that. Thanks

File: README.md (hunk 1 above)
Comment (naveensrinivasan) [reply README.md]: I included a reference to Sigstore and other components.

File: README.md [hunk 4]
```diff
@@ -254,3 +256,32 @@ jobs:
         with:
           sarif_file: results.sarif
 ```
+## Verifying the Authenticity and Integrity of Scorecard Results
+The scorecard system uses several components from [sigstore](https://github.com/sigstore) to ensure tamper-resistant results. These components include cosign, fulcio, and rekor.
+
+Cosign helps to sign the scorecard action results with a cryptographic signature and stores the hash in a tamper-proof store called rekor, which uses technology based on Merkel trees (a specific type of data structure used in blockchain systems).
```
Comment (olivekl) [inline README.md:262]:
```suggestion
Cosign signs the Scorecard action results with a cryptographic signature and stores the hash in a tamper-proof store called rekor, which uses technology based on Merkel trees (a specific type of data structure used in blockchain systems).
```

File: README.md [hunk 5]
```diff
@@ -254,3 +256,32 @@ jobs:
...
+
+Fulcio is a free certificate authority that issues short-lived code signing certificates for OpenID Connect identities, such as email addresses. Cosign uses fulcio for ephemeral keys and certificates.
+
+Rekor is a system that aims to provide an unchanging ledger of metadata generated within a software project's supply chain. It allows software maintainers and build systems to record signed metadata, which is then recorded in an immutable record using Merkel trees. Other parties can use this metadata to determine if they can trust the lifecycle of an object. Cosign stores the signed hash of the result in rekor to further enhance the tamper resistance of the scorecard system.
```
Comment (olivekl) [inline README.md:266]:
```suggestion
Rekor is a system that aims to provide an unchanging ledger of metadata generated within a software project's supply chain. It allows software maintainers and build systems to record signed metadata, which is then recorded in an immutable record using Merkel trees. Other parties can use this metadata to determine if they can trust the lifecycle of an object. Cosign stores the signed hash of the result in rekor to further enhance the tamper resistance of the Scorecard system.
```

File: README.md [hunk 6]
```diff
@@ -254,3 +256,32 @@ jobs:
...
+
+You can learn more about Merkel trees at this link: https://en.wikipedia.org/wiki/Merkle_tree
+
+
+This diagram illustrates how Scorecard validates the authenticity and integrity of the results it produces.This process allows the repository owner to verify the authenticity and integrity of the scorecard results and make them available for others to access and use.
```
Comment (olivekl) [inline README.md:271]:
```suggestion
This diagram illustrates how Scorecard validates the authenticity and integrity of the results it produces. This process allows the repository owner to verify the authenticity and integrity of Scorecard results and make them available for others to access and use.
```

**Review by olivekl — APPROVED** [review]
Minor changes, LGTM

**Comment by naveensrinivasan** [comment]
> Minor changes, LGTM

Thanks @olivekl  for making it better! Appreciate the detailed feedback and suggestion!

File: README.md [hunk 7]
```diff
@@ -254,3 +256,32 @@ jobs:
...
 ```
+## Verifying the Authenticity and Integrity of Scorecard Results
+The scorecard system uses several components from [sigstore](https://github.com/sigstore) to ensure tamper-resistant results. These components include cosign, fulcio, and rekor.
+
+Cosign signs the Scorecard action results with a cryptographic signature and stores the hash in a tamper-proof store called rekor, which uses technology based on Merkel trees (a specific type of data structure used in blockchain systems).
+
+Fulcio is a free certificate authority that issues short-lived code signing certificates for OpenID Connect identities, such as email addresses. Cosign uses fulcio for ephemeral keys and certificates.
+
+Rekor is a system that aims to provide an unchanging ledger of metadata generated within a software project's supply chain. It allows software maintainers and build systems to record signed metadata, which is then recorded in an immutable record using Merkel trees. Other parties can use this metadata to determine if they can trust the lifecycle of an object. Cosign stores the signed hash of the result in rekor to further enhance the tamper resistance of the Scorecard system.
```
Comment (laurentsimon) [inline README.md:266]: nit: unchanging -> immutable

Comment (laurentsimon) [inline README.md:266]: You could just say "is a transparency logs" for signatures, and link to their repo or some Sigstore's official doc.

File: README.md [hunk 8]
```diff
@@ -254,3 +256,32 @@ jobs:
...
+
+You can learn more about Merkel trees at this link: https://en.wikipedia.org/wiki/Merkle_tree
+
```
Comment (laurentsimon) [inline README.md:269]: there is a simplification you've made that people will point out is insecure. We don't just validate a signature: we validate the code of the workflow that produced the result. Signing something does not ensure that the code that ran is the right one.

**Review by laurentsimon — COMMENTED** [review]
see inline comments

File: README.md [hunk 9]
```diff
@@ -40,6 +40,8 @@ ________
 - [Publishing Results](#publishing-results)
 - [Uploading Artifacts](#uploading-artifacts)
 - [Workflow Example](#workflow-example)
+[Verifying the Authenticity and Integrity of Scorecard Results](#verifying-the-authenticity-and-integrity-of-scorecard-results)
```
Comment (spencerschrock) [inline README.md:43]: missing a `-` so the list breaks

**Review by spencerschrock — COMMENTED** [review]
@naveensrinivasan Do you still plan to work on this? I will be writing a closely related section to deal with #1150, and would like this to be merged

**Pull request closed without merging**